core/templates/dev/head/pages/topics_and_skills_dashboard/SkillsListDirective.js

```diff
@@ -129,6 +136,64 @@ oppia.directive('skillsList', [
               }
             });
           };
+
+          $scope.selectSkill = function(skill) {
+            $scope.selectedSkill = skill;
+          };
+
+          $scope.mergeSkill = function(skill) {
+            var skillSummaries = $scope.getMergeableSkillSummaries();
+            var modalInstance = $uibModal.open({
+              templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                '/pages/topics_and_skills_dashboard/' +
+                'merge_skill_modal_directive.html'),
+              backdrop: true,
+              controller: [
+                '$scope', '$uibModalInstance',
+                function($scope, $uibModalInstance) {
+                  $scope.skillSummaries = skillSummaries;
+                  $scope.selectedSkill = {};
+                  $scope.done = function() {
+                    $uibModalInstance.close(
+                      {skill: skill,
+                        supersedingSkill: $scope.selectedSkill
+                      });
+                  };
+                  $scope.cancel = function() {
+                    $uibModalInstance.dismiss('cancel');
+                  };
+                }
+              ]
+            });
+
+            modalInstance.result.then(function(result) {
+              var skill = result.skill;
+              var supersedingSkill = result.supersedingSkill;
+              var changeList = [{
+                cmd: 'update_skill_property',
+                property_name: 'superseding_skill_id',
+                old_value: '',
+                new_value: supersedingSkill.id
+              }];
+              EditableSkillBackendApiService.updateSkill(
+                skill.id, skill.version,
+                'Added superseding skill id ' +
+                 supersedingSkill.id + ' to skill.',
+                changeList
+              ).then(function() {
+                $rootScope.$broadcast(
```

Comment:
Should you also be updating the frontend objects as well, or will that happen later automatically? Maybe this is the point at which you use setSupersedingSkillId() etc.

Reply:
Done